### PR TITLE
TF-1432 Fix all child/grandchild mailboxes don't hide themselves when hiding parent mailbox

### PR DESCRIPTION
**Pull request by dab246**
#1432 

### Resolved

- Mobile

https://user-images.githubusercontent.com/80730648/218031414-0d2ae7ea-59c5-4e80-ba46-53bbc3a73e09.mp4

- Web 

https://user-images.githubusercontent.com/80730648/218031576-9ad66528-4277-4fbd-85f3-3caafcb67627.mov

### CODE REVIEW

**Comment by hoangdat** [comment]
hi @dab246 , please demo the case.

1. select mailbox A
2. all mails inside A are displayed
3. open an email
4. hide mailbox A

File: lib/features/mailbox/domain/extensions/list_mailbox_id_extension.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:jmap_dart_client/jmap/core/patch_object.dart';
+import 'package:jmap_dart_client/jmap/mail/mailbox/mailbox.dart';
+import 'package:model/mailbox/mailbox_property.dart';
+import 'package:tmail_ui_user/features/mailbox/domain/model/mailbox_subscribe_state.dart';
+
+extension ListMailboxIdExtension on List<MailboxId> {
+
+  Map<Id, PatchObject> generateMapUpdateObjectSubscribeMailbox(MailboxSubscribeState subscribeState) {
+    final Map<Id, PatchObject> maps = {};
+    forEach((mailboxId) {
```
Comment (hoangdat) [inline lib/features/mailbox/domain/extensions/list_mailbox_id_extension.dart:11]: why not use `map` instead of `forEach`?

Comment (dab246) [reply lib/features/mailbox/domain/extensions/list_mailbox_id_extension.dart]: You can use different ways to convert `List` to `Map`. Example: 
```
var result = Map.fromIterable(l, key: (v) => v[0], value: (v) => v[1]);
```

But from starting from `Dart 2.3`, we are encouraged to use `collection-for`.
```
var result = { for (var v in l) v[0]: v[1] };
```

So I converted `forEach` to `for-in`. See details in `commit-fixup`

**Comment by dab246** [comment]
> hi @dab246 , please demo the case.
> 
> 1. select mailbox A
> 2. all mails inside A are display
> 3. hide mailbox A



> hi @dab246 , please demo the case.
> 
> 1. select mailbox A
> 2. all mails inside A are displayed
> 3. open an email
> 4. hide mailbox A

@hoangdat  Please see the demo.

https://user-images.githubusercontent.com/80730648/218081682-cb86b95a-ef66-432d-9382-25589579414d.mov

File: lib/features/mailbox/data/network/mailbox_api.dart [hunk 2]
```diff
@@ -321,4 +324,34 @@ class MailboxAPI with HandleSetErrorMixin {
       throw error;
     });
   }
+
+  Future<List<MailboxId>> subscribeMultipleMailbox(AccountId accountId, SubscribeMultipleMailboxRequest subscribeRequest) async {
+    final mapMailboxUpdated = subscribeRequest.mailboxIdsSubscribe
+      .generateMapUpdateObjectSubscribeMailbox(subscribeRequest.subscribeState);
```
Comment (hoangdat) [inline lib/features/mailbox/data/network/mailbox_api.dart:330]: We dont need unsubcribe children first?

**Review by nqhhdev — APPROVED** [review]
Lgtm 🚀

File: lib/features/mailbox/data/network/mailbox_api.dart (hunk 2 above)
Comment (dab246) [reply lib/features/mailbox/data/network/mailbox_api.dart:330]: That's right.

**Pull request merged**